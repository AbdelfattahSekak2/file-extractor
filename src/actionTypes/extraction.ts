export const GetExtraction = {
  FAILURE: "GET_EXTRACTION_FAILURE",
  SUCCESS: "GET_EXTRACTION_SUCCESS",
  REQUEST: "GET_EXTRACTION_REQUEST",
}

export const ClearExtractionState = {
  DEFAULT: "CLEAR_EXTRACTION_STATE_DEFAULT",
}
