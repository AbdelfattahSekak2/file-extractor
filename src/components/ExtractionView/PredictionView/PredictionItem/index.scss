@import "../../../../styles/mixins";

.prediction-item {
  display: flex;
  flex-direction: row;
  border-radius: $border-radius;
  margin-top: 30px;
  box-shadow: $box-shadow;
  label {
    color: $dark-grey;
    background: white;
    padding: 2px 10px;
    border-radius: $border-radius;
    width: 130px;
    font-size: 15px;
    font-weight: 500;
    line-height: 40px;
  }
  .details {
    position: relative;
    width: 100%;
    height: 100%;
    align-self: center;
    z-index: 2;
    background: #bbbaba;
    border-radius: 0 $border-radius $border-radius 0;
    .bar {
      z-index: -1;
      position: absolute;
      top: 0;
      background: $mindee-red;
      width: 100%;
      height: 100%;
    }
    p {
      padding: 5px;
      margin: 10px;
      line-height: 10px;
      position: unset !important;
      z-index: 1;
      color: rgb(255, 254, 254);
      text-align: center;
    }
  }
  .details-list {
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100%;
    align-self: center;
    background: none;
    .details + .details {
      margin-top: 5px;
    }
  }
}
