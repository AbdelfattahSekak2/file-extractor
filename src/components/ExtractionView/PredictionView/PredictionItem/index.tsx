import * as React from "react";

import "./index.scss";

interface Element {
  value: string | number;
  sign?: string;
  subValue?: {
    label: string;
    value: string | number;
  }
  probability: number;
}

interface IProps {
  label: string;
  element: Element | Element[];
}

export const getProbability = (probability: number) => ({
    width: `${probability * 100}%`,
    borderRadius: probability === 1 ? "0 6px 6px 0" : "0",
  })

const SubValue = ({ value, label }: { value: string | number; label: string }) => <>
  ({label} : <strong>{value}</strong>)
</>;

const PredictionItem: React.FunctionComponent<IProps> = ({
  label,
  element,
}) => {
  return <div className="prediction-item">
    <label>{label}</label>
    {
      Array.isArray(element)
        ? <div className="details-list">
          {
            element.map((item: Element, key: number) =>
              <div className="details" key={key} >
                <p>
                  {item.value}{item.sign} {item.subValue ? <SubValue {...item.subValue} /> : null}
                </p>
                <div className="bar" style={getProbability(item.probability)} />
              </div>
            )
          }
        </div>
        : <div className="details">
          <p>{element.value}{element.sign}</p>
          <div className="bar" style={getProbability(element.probability)} />
        </div>
    }
  </div>;
};

export default PredictionItem;

