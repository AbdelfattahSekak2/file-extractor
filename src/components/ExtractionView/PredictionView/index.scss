.prediction-container {
  margin-right: 10px;
  display: flex;
  flex-direction: column;
  width: 100%;
  .title {
    color: #aaa7a7;
    font-weight: 700;
    font-size: 22px;
    margin: 10px;
    text-align: center;
  }
}
