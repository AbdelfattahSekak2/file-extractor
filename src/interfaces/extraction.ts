export interface Coordinates {
  coord_x1: number;
  coord_x2: number;
  coord_y1: number;
  coord_y2: number;
}

export interface Tax extends Coordinates {
  rate: number;
  amount: number;
  probability: number;
}

interface Date extends Coordinates {
  date_INTL: string,
  date_US: string,
  probability: 0.972,
}

interface Total extends Coordinates {
  amount: number,
  probability: number,
}

export interface Prediction {
  orientation: {
    degrees: string;
    probability: number;
  },
  category: {
    value: string;
    probability: number;
  },
  date: Date
  total: Total;
  taxes: Tax[]
}

export default interface Extraction {
  details: {
    call_id: string;
    computation_time: number;
  };
  predictions: Prediction[];
}
