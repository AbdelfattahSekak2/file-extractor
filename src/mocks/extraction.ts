import Extraction from "../interfaces/extractions";

export const extraction: Extraction = {
  "details": {
    "call_id": "49ef6b42-2724-4eaf-80bf-9add06a45176",
    "computation_time": 0.789,
  },
  "predictions": [
    {
      "orientation": {
        "degrees": "0",
        "probability": 0.723
      },
      "category": {
        "value": "parking",
        "probability": 0.302
      },
      "date": {
        "date_INTL": "2018-01-02",
        "date_US": "2012-10-26",
        "probability": 0.972,
        "coord_x1": 0.865,
        "coord_x2": 0.18,
        "coord_y1": 0.249,
        "coord_y2": 0.818
      },
      "total": {
        "amount": 0.732,
        "probability": 0.801,
        "coord_x1": 0.255,
        "coord_x2": 0.2,
        "coord_y1": 0.461,
        "coord_y2": 0.927,
      },
      "taxes": [
        {
          "rate": 0.476,
          "amount": 0.57,
          "probability": 0.647,
          "coord_x1": 0.216,
          "coord_x2": 0.256,
          "coord_y1": 0.16,
          "coord_y2": 0.145,
        }
      ]
    },
  ]
}
