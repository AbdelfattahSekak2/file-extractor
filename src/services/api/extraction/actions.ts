import { ClearExtractionState, GetExtraction } from "../../../actionTypes/extraction";

export function getExtraction(file: FormData) {
  return {
    file,
    type: GetExtraction.REQUEST,
  };
}

export function clearExtractionState() {
  return {
    type: ClearExtractionState.DEFAULT,
  };
}
