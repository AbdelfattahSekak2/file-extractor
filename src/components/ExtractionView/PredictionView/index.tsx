import * as React from "react";

import { Prediction, Tax } from "../../../interfaces/extraction";

import "./index.scss";
import PredictionItem from "./PredictionItem";

interface IProps {
  prediction: Prediction;
}

const PredictionView: React.FunctionComponent<IProps> = ({
  prediction,
}) => {
  const { orientation, taxes, date, total, category } = prediction;
  return (
    <div className="prediction-container" >
      {
        orientation &&
        <PredictionItem
          label="Orientation"
          element={{
            probability: orientation.probability,
            sign: "°",
            value: orientation.degrees,
          }}
        />
      }
      {
        category &&
        <PredictionItem
          label="Category"
          element={{
            probability: category.probability,
            value: category.value,
          }}
        />
      }
      {
        date &&
        <PredictionItem
          label="Date"
          element={{
            probability: date.probability,
            value: date.date_INTL,
          }}
        />
      }
      {
        total &&
        <PredictionItem
          label="Total"
          element={{
            probability: total.probability,
            sign: "€",
            value: total.amount,
          }}
        />
      }
      {
        taxes && taxes.length > 0 &&
        <PredictionItem
          label="Taxes"
          element={taxes.map((e: Tax) => ({
            probability: e.probability,
            sign: "€",
            subValue: {
              label: "Rate",
              value: e.rate,
            },
            value: e.amount,
          }))}
        />
      }
    </div >
  );
};

export default PredictionView;
