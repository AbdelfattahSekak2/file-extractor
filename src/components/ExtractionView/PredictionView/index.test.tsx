import React from "react";
import { ShallowWrapper, shallow } from "enzyme";

import PredictionView from "."

import { extraction } from "../../../mocks/extraction";

describe("PredictionView  component", () => {
  let wrapper: ShallowWrapper;
  const prediction: any = extraction.predictions[0];
  it("should render correctly", () => {
    wrapper = shallow(<PredictionView count={1} prediction={prediction} />);
    expect(wrapper).not.toBeNull();
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot when total is undefined", () => {
    wrapper = shallow(<PredictionView count={1} prediction={{
      ...prediction,
      total: undefined,
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should match snapshot when category is undefined", () => {
    wrapper = shallow(<PredictionView count={1} prediction={{
      ...prediction,
      category: undefined,
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should match snapshot when date is undefined", () => {
    wrapper = shallow(<PredictionView count={1} prediction={{
      ...prediction,
      date: undefined,
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should match snapshot when orientation is undefined", () => {
    wrapper = shallow(<PredictionView count={1} prediction={{
      ...prediction,
      orientation: undefined,
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should match snapshot when taxes is undefined", () => {
    wrapper = shallow(<PredictionView count={1} prediction={{
      ...prediction,
      taxes: undefined,
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should match snapshot when taxes is empty", () => {
    wrapper = shallow(<PredictionView count={1} prediction={{
      ...prediction,
      taxes: [],
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
})
