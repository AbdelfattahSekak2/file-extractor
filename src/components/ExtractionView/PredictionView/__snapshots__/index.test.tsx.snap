// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictionView  component should match snapshot when category is undefined 1`] = `
<div
  className="prediction-container"
>
  <PredictionItem
    element={
      Object {
        "probability": 0.723,
        "sign": "°",
        "value": "0",
      }
    }
    label="Orientation"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.972,
        "value": "2018-01-02",
      }
    }
    label="Date"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.801,
        "sign": "€",
        "value": 0.732,
      }
    }
    label="Total"
  />
  <PredictionItem
    element={
      Array [
        Object {
          "probability": 0.647,
          "sign": "€",
          "subValue": Object {
            "label": "Rate",
            "value": 0.476,
          },
          "value": 0.57,
        },
      ]
    }
    label="Taxes"
  />
</div>
`;

exports[`PredictionView  component should match snapshot when date is undefined 1`] = `
<div
  className="prediction-container"
>
  <PredictionItem
    element={
      Object {
        "probability": 0.723,
        "sign": "°",
        "value": "0",
      }
    }
    label="Orientation"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.302,
        "value": "parking",
      }
    }
    label="Category"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.801,
        "sign": "€",
        "value": 0.732,
      }
    }
    label="Total"
  />
  <PredictionItem
    element={
      Array [
        Object {
          "probability": 0.647,
          "sign": "€",
          "subValue": Object {
            "label": "Rate",
            "value": 0.476,
          },
          "value": 0.57,
        },
      ]
    }
    label="Taxes"
  />
</div>
`;

exports[`PredictionView  component should match snapshot when orientation is undefined 1`] = `
<div
  className="prediction-container"
>
  <PredictionItem
    element={
      Object {
        "probability": 0.302,
        "value": "parking",
      }
    }
    label="Category"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.972,
        "value": "2018-01-02",
      }
    }
    label="Date"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.801,
        "sign": "€",
        "value": 0.732,
      }
    }
    label="Total"
  />
  <PredictionItem
    element={
      Array [
        Object {
          "probability": 0.647,
          "sign": "€",
          "subValue": Object {
            "label": "Rate",
            "value": 0.476,
          },
          "value": 0.57,
        },
      ]
    }
    label="Taxes"
  />
</div>
`;

exports[`PredictionView  component should match snapshot when taxes is empty 1`] = `
<div
  className="prediction-container"
>
  <PredictionItem
    element={
      Object {
        "probability": 0.723,
        "sign": "°",
        "value": "0",
      }
    }
    label="Orientation"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.302,
        "value": "parking",
      }
    }
    label="Category"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.972,
        "value": "2018-01-02",
      }
    }
    label="Date"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.801,
        "sign": "€",
        "value": 0.732,
      }
    }
    label="Total"
  />
</div>
`;

exports[`PredictionView  component should match snapshot when taxes is undefined 1`] = `
<div
  className="prediction-container"
>
  <PredictionItem
    element={
      Object {
        "probability": 0.723,
        "sign": "°",
        "value": "0",
      }
    }
    label="Orientation"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.302,
        "value": "parking",
      }
    }
    label="Category"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.972,
        "value": "2018-01-02",
      }
    }
    label="Date"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.801,
        "sign": "€",
        "value": 0.732,
      }
    }
    label="Total"
  />
</div>
`;

exports[`PredictionView  component should match snapshot when total is undefined 1`] = `
<div
  className="prediction-container"
>
  <PredictionItem
    element={
      Object {
        "probability": 0.723,
        "sign": "°",
        "value": "0",
      }
    }
    label="Orientation"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.302,
        "value": "parking",
      }
    }
    label="Category"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.972,
        "value": "2018-01-02",
      }
    }
    label="Date"
  />
  <PredictionItem
    element={
      Array [
        Object {
          "probability": 0.647,
          "sign": "€",
          "subValue": Object {
            "label": "Rate",
            "value": 0.476,
          },
          "value": 0.57,
        },
      ]
    }
    label="Taxes"
  />
</div>
`;

exports[`PredictionView  component should render correctly 1`] = `
<div
  className="prediction-container"
>
  <PredictionItem
    element={
      Object {
        "probability": 0.723,
        "sign": "°",
        "value": "0",
      }
    }
    label="Orientation"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.302,
        "value": "parking",
      }
    }
    label="Category"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.972,
        "value": "2018-01-02",
      }
    }
    label="Date"
  />
  <PredictionItem
    element={
      Object {
        "probability": 0.801,
        "sign": "€",
        "value": 0.732,
      }
    }
    label="Total"
  />
  <PredictionItem
    element={
      Array [
        Object {
          "probability": 0.647,
          "sign": "€",
          "subValue": Object {
            "label": "Rate",
            "value": 0.476,
          },
          "value": 0.57,
        },
      ]
    }
    label="Taxes"
  />
</div>
`;
